pkg/gofr/datasource/pubsub/google/google_test.go

```diff
@@ -174,17 +175,17 @@ func Test_validateConfigs(t *testing.T) {
 }
 
 func TestGoogleClient_CloseReturnsError(t *testing.T) {
-	// client already present
 	g := &googleClient{
-		client: getGoogleClient(t),
+		client:      getGoogleClient(t),
+		receiveChan: make(map[string]chan *pubsub.Message),
 	}
 
 	err := g.Close()
 
 	require.NoError(t, err)
 
 	// client empty
-	g = &googleClient{}
+	g = &googleClient{receiveChan: make(map[string]chan *pubsub.Message)}
```

Comment:
I tend to define the things like that in newConfig function you can pass parameters or an Init method on the googleClient

This way the private data prerequisites are not spread in multiple files